### ISSUE TITLE
mGBA 10.0.3 (Wii port) will not work on startup

### ISSUE THREAD

**Issue by BlackPaladin2022** [mgba-emu/mgba#3106]
I recently download the most recent version of mGBA to try it out (0.10.3), but when I loaded the emulator, the program will not run at all.  All I get is a blank screen.  (I have to press the reset button to get out of the program itself.)

This was not what I expected.

**Comment by faythdus** [comment]
The Psvita port not work too

**Comment by saulfabregwiivc** [comment]
Try deletung your old config files?

**Comment by saulfabregwiivc** [comment]
Welp, i have the exact same issue, but this time i get a DSI exception:

![image](https://github.com/mgba-emu/mgba/assets/66485640/308f9158-a02f-4ff4-ae2a-77c174572d49)

**Comment by endrift** [comment]
I don't know why it's crashing now, especially only for some people (it doesn't crash for me), but it looks from the crash dump above that it might have to do with how many or which devices you have plugged in.

**Comment by endrift** [comment]
I haven't been able to reproduce yet, but can you see if this fixes anything?

[mgba-vita.zip](https://github.com/mgba-emu/mgba/files/14000577/mgba-vita.zip)
[mgba-wii.zip](https://github.com/mgba-emu/mgba/files/14000579/mgba-wii.zip)

**Comment by BlackPaladin2022** [comment]
I tested out that .DOL file on my Wii... and it worked!  I don't know what you did to it, but that appeared to have did the trick.  Thank you so much!

**Comment by saulfabregwiivc** [comment]
@endrift Tested - yep, the new DOL fixed the problem, thank you so much!

![image](https://github.com/mgba-emu/mgba/assets/66485640/64a14250-ae7f-46ea-afee-8c76e8a616fb)

Will you commit the new fix on upstream?

**Comment by endrift** [comment]
Yeah, I committed the fixed version yesterday. It wasn't clear to me if it'd fix the issue but it was a definite improvement to code quality so I committed it before I confirmed if it fixed it: https://github.com/mgba-emu/mgba/commit/6e773eafeae96c8904cf6e6de0b3f8a3b33f4235

I'd like confirmation from @BlackPaladin2022 before I close this though.

**Comment by BlackPaladin2022** [comment]
@endrift I did test it out on my WII, and that version of the .DOL file did work!  With that, I can use mGBA on my Wii.  Thank you so much for fixing this!  :)

**Comment by endrift** [comment]
Good to hear it's fixed. I have barely any idea why that fixed it but I'll take it.

**Comment by DaniielBrew** [comment]
THANK YOU SO MUCH"
it work so well!
:)

**Comment by DaniielBrew** [comment]
incredible
